references/validationFunctions/treatment/checkWhenNoTreatment.js

```diff
@@ -23,63 +23,63 @@
  */
 
 const validation = () =>
-    (function validate(inputs) {
-        const { $row, $name, $field } = inputs;
-        const result = { valid: true, message: 'Ok' };
+	function validate(inputs) {
+		const { $row, $name, $field } = inputs;
+		const result = { valid: true, message: 'Ok' };
 
-        const arrayItemsInSecondArray = (arr1, arr2) => {
-            return arr2.some(arr2Item => {
-                return arr1.includes(arr2Item);
-            });
-        };
+		const arrayItemsInSecondArray = (arr1, arr2) => {
+			return arr2.some(arr2Item => {
+				return arr1.includes(arr2Item);
+			});
+		};
 
-        const coreFields = [
-            'treatment_start_interval',
-            'treatment_duration',
-            'is_primary_treatment',
-            'treatment_intent',
-            'treatment_setting',
-            'response_to_treatment_criteria_method',
-            'response_to_treatment',
-        ];
+		const coreFields = [
+			'treatment_start_interval',
+			'treatment_duration',
+			'is_primary_treatment',
+			'treatment_intent',
+			'treatment_setting',
+			'response_to_treatment_criteria_method',
+			'response_to_treatment',
+		];
 
-        const treatmentExceptionTypes = ['no treatment', 'unknown'];
+		const treatmentExceptionTypes = ['no treatment', 'unknown'];
 
-        // checks for a string just consisting of whitespace
-        const checkforEmpty = entry => {
-            return /^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'));
-        };
-        const treatmentTypes = $row.treatment_type.map(value => value.toLowerCase());
+		const checkforInvalid = entry => {
+			const entryAsNumber = parseFloat($field); // to validate it as a numnber
 
-        const recordHasTreatments = !arrayItemsInSecondArray(
-            treatmentExceptionTypes,
-            treatmentTypes,
-        );
+			return (
+				[null, undefined, ''].includes($field) || // first dispose of the regular falsy values
+				(!isNaN(entryAsNumber) && entryAsNumber < 0) || // checks for valid numbers
+				/^\s+$/g.test(decodeURI(entry).replace(/^"(.*)"$/, '$1'))
+			); // whitespace-filled strings
+		};
```

Comment:
This function now checks beyond whitespace strings, also whether there's `falsy` values, while allowing `0+` to pass.

Reply:
sorry @justincorrigible I forgot to mention that the intervals have been adjusted to accept negative numbers: https://github.com/icgc-argo/argo-dictionary/issues/436

Reply:
ahh! good to know! the code should reflect that 👍 
I'll make the changes and document as a comment for future reference
(using explanation in https://github.com/icgc-argo/argo-dictionary/issues/432)